internal/printer/printer.go

```diff
@@ -291,11 +298,15 @@ func (p *printer) printAttribute(attr astro.Attribute) {
 		p.print(strings.TrimSpace(attr.Key))
 		p.print(`, "` + strings.TrimSpace(attr.Key) + `")}`)
 	case astro.ShorthandAttribute:
+		withoutComments := removeComments(attr.Key)
+		if len(withoutComments) == 0 {
+			return
+		}
 		p.print(fmt.Sprintf("${%s(", ADD_ATTRIBUTE))
 		p.addSourceMapping(attr.KeyLoc)
 		p.print(strings.TrimSpace(attr.Key))
 		p.addSourceMapping(attr.KeyLoc)
-		p.print(`, "` + strings.TrimSpace(attr.Key) + `")}`)
+		p.print(`, "` + withoutComments + `")}`)
```

Comment:
This and [`ln 208`](https://github.com/withastro/compiler/pull/364/files#diff-58de1575bcd3abab85340eb61810590c68a061ec2cad49d8cf3864fe402a64c1L208) would resolve the key for shorthand attribute with comments like so `{/* a comment */ id} => { "/* a comment */ id":  id }`

We remove the comments now when printing the key to fix this